src/components/view-pane/related-views-button.tsx

```diff
@@ -18,25 +22,38 @@ export class RelatedViewsButtonBase extends React.PureComponent<RelatedViewsButt
   }
 
   public render() {
+    const {config, relatedViews} = this.props;
+    const {hideRelatedViews} = config;
+    const hideCondition = (relatedViews.isHidden === undefined && hideRelatedViews) ||
+      (relatedViews.isHidden !== undefined && relatedViews.isHidden);
     return (
       <div styleName="right">
         <a onClick={this.onHideClick}>
-          {this.props.relatedViews.isHidden ? 'Show' : 'Hide'}
+          {hideCondition ? "Show" : "Hide"}
```

Comment:
for JS parts, use single quotes -- not sure why you have to change to double quotes.